Upgrading to @nativescript/webpack-5.0.19 requires upgrade to @nativescript/core-8.7.0

### Issue Description

I'm currently using @nativescript/webpack 5.0.18 with @nativescript/core 8.3.6 (yes, I know that's old). Everything builds fine. However, when I upgrade @nativescript/webpack to 5.0.19 my Android build breaks.

The issue seems to trace to [this PR](https://github.com/NativeScript/NativeScript/pull/10506/files#diff-55d8fcafa646ec0c0877526af8ff20e1a53ab0b0fd7f2506aa10f67e3a98ecf1), specifically [this line](https://github.com/NativeScript/NativeScript/blob/2b0f2678bad7b36172ac84bcb0b1dca074786562/packages/webpack5/src/configuration/base.ts#L521). 

I would not have expected to have to do a significant upgrade of @nativescript/core (a runtime dep) to take a patch version of @nativescript/webpack (a devDep), so I logged this as a bug, but maybe it's just how it is. Still, I'd expect @nativescript/webpack to at least get a minor version bump in this scenario rather than what's supposed to be a safe "patch".

### Reproduction

1. Use @nativescript/webpack 5.0.19 w/ a version of @nativescript/core prior to 8.7.0.
2. Execute `ns build android`

### Relevant log output (if applicable)

```shell
ERROR in tns_modules/inspector_modules
Module not found: Error: Can't resolve '@nativescript/core/inspector_modules' in '...'
resolve '@nativescript/core/inspector_modules' in '...'
  Parsed request is a module
  using description file: .../package.json (relative path: .)
    resolve as module
      looking for modules in .../node_modules
        existing directory .../node_modules/@nativescript/core
          using description file: .../node_modules/@nativescript/core/package.json (relative path: .)
            using description file: .../node_modules/@nativescript/core/package.json (relative path: ./inspector_modules)
              no extension
                .../node_modules/@nativescript/core/inspector_modules doesn't exist
              .android.ts
                .../node_modules/@nativescript/core/inspector_modules.android.ts doesn't exist
              .ts
                .../node_modules/@nativescript/core/inspector_modules.ts doesn't exist
              .android.js
                .../node_modules/@nativescript/core/inspector_modules.android.js doesn't exist
              .js
                .../node_modules/@nativescript/core/inspector_modules.js doesn't exist
              .android.mjs
                .../node_modules/@nativescript/core/inspector_modules.android.mjs doesn't exist
              .mjs
                .../node_modules/@nativescript/core/inspector_modules.mjs doesn't exist
              .android.css
                .../node_modules/@nativescript/core/inspector_modules.android.css doesn't exist
              .css
                .../node_modules/@nativescript/core/inspector_modules.css doesn't exist
              .android.scss
                .../node_modules/@nativescript/core/inspector_modules.android.scss doesn't exist
              .scss
                .../node_modules/@nativescript/core/inspector_modules.scss doesn't exist
              .android.json
                .../node_modules/@nativescript/core/inspector_modules.android.json doesn't exist
              .json
                .../node_modules/@nativescript/core/inspector_modules.json doesn't exist
              as directory
                .../node_modules/@nativescript/core/inspector_modules doesn't exist


ls node_modules/@nativescript/core/inspector_modules*
----------------
node_modules/@nativescript/core/inspector_modules.ios.d.ts
node_modules/@nativescript/core/inspector_modules.ios.js
node_modules/@nativescript/core/inspector_modules.ios.js.map
```


### Environment

npm WARN exec The following package was not found and will be installed: nativescript-envinfo@1.0.9
<!-- COPY START -->
```yaml
OS: macOS 14.4.1
CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
Shell: /bin/zsh
node: 18.19.0
npm: 10.2.3
nativescript: 8.7.0

# android
java: 17.0.9
ndk: Not Found
apis: 33
build_tools: 33.0.2
system_images: Not Found

# ios
xcode: 15.3/15E204a
cocoapods: 1.15.2
python: Not Found
python3: 3.11.7
ruby: 2.7.8
platforms:
  - DriverKit 23.4
  - iOS 17.4
  - macOS 14.4
  - tvOS 17.4
  - visionOS 1.1
  - watchOS 10.4
```

### Dependencies

```json
"dependencies": {
  "@angular/animations": "~12.0.0",
  "@angular/common": "~12.0.0",
  "@angular/compiler": "~12.0.0",
  "@angular/core": "~12.0.0",
  "@angular/forms": "~12.0.0",
  "@angular/platform-browser": "~12.0.0",
  "@angular/platform-browser-dynamic": "~12.0.0",
  "@angular/router": "~12.0.0",
  "@nativescript-community/drawingpad": "~4.1.0",
  "@nativescript-community/ui-material-activityindicator": "~7.2.0",
  "@nativescript-community/ui-material-button": "~7.2.0",
  "@nativescript-community/ui-material-tabs": "~7.2.0",
  "@nativescript-community/ui-material-textfield": "~7.2.0",
  "@nativescript/angular": "~12.0.0",
  "@nativescript/core": "~8.3.5",
  "@nativescript/datetimepicker": "~2.1.9",
  "@nativescript/directions": "~2.0.1",
  "@nativescript/geolocation": "~8.3.0",
  "@nativescript/theme": "~3.0.1",
  "@nativescript/unit-test-runner": "~2.0.5",
  "@ngrx/effects": "~12.5.0",
  "@ngrx/store": "~12.5.0",
  "@nstudio/nativescript-audio-recorder": "~0.1.0",
  "@nstudio/nativescript-checkbox": "~2.0.4",
  "material-design-icons": "~3.0.1",
  "moment": "~2.29.3",
  "nativescript-audio": "~6.2.6",
  "nativescript-fonticon": "~2.0.2",
  "nativescript-modal-datetimepicker": "~2.1.5",
  "nativescript-ng2-fonticon": "~1.3.4",
  "nativescript-phone": "~3.0.2",
  "nativescript-speech-recognition": "~2.0.0",
  "nativescript-ui-listview": "~10.0.2",
  "nativescript-ui-sidedrawer": "~10.0.1",
  "rxjs": "~6.6.0",
  "zone.js": "~0.11.4"
},
"devDependencies": {
  "@angular/compiler-cli": "~12.0.0",
  "@nativescript/android": "~8.2.2",
  "@nativescript/ios": "~8.3.3",
  "@nativescript/types": "~8.2.0",
  "@nativescript/webpack": "~5.0.6",
  "@ngtools/webpack": "~12.0.0",
  "@types/jasmine": "~3.7.2",
  "karma": "~6.3.17",
  "karma-jasmine": "~4.0.1",
  "karma-nativescript-launcher": "~0.4.0",
  "karma-webpack": "~5.0.0",
  "sass": "~1.49.9",
  "sass-loader": "~12.6.0",
  "typescript": "~4.2.0"
}
```
<!-- COPY END -->

### Please accept these terms

- [X] I have searched the [existing issues](https://github.com/NativeScript/NativeScript/issues) as well as [StackOverflow](https://stackoverflow.com/questions/tagged/nativescript) and this has not been posted before
- [X] This is a bug report
- [X] I agree to follow this project's [Code of Conduct](https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#coc)

--- a/packages/webpack5/__tests__/configuration/base.spec.ts
+++ b/packages/webpack5/__tests__/configuration/base.spec.ts
@@ -4,6 +4,7 @@ import fs from 'fs';
 import base from '../../src/configuration/base';
 import { init } from '../../src';
 import { applyFileReplacements } from '../../src/helpers/fileReplacements';
+import * as dependenciesHelpers from '../../src/helpers/dependencies';
 import { additionalCopyRules } from '../../src/helpers/copyRules';
 
 describe('base configuration', () => {
@@ -220,5 +221,31 @@ describe('base configuration', () => {
 
 		expect(config.output.get('sourceMapFilename')).toMatchSnapshot();
 		expect(config.get('devtool')).toBe('hidden-source-map');
+	});
+
+	it('includes inspector_modules on android when @nativescript/core version is >= 8.7.0', () => {
+		const getDependencyVersionSpy = jest.spyOn(
+			dependenciesHelpers,
+			'getDependencyVersion'
+		);
+		getDependencyVersionSpy.withImplementation(
+			(name) => {
+				if (name === '@nativescript/core') {
+					return '8.7.0';
+				}
+				return null;
+			},
+			() => {
+				init({
+					android: true,
+				});
+
+				const config = base(new Config());
+				const entry = config.entryPoints.get('tns_modules/inspector_modules');
+
+				expect(entry).toBeDefined();
+				expect(entry.values().length).toBe(1);
+			}
+		);
 	});
 });

--- a/packages/webpack5/package.json
+++ b/packages/webpack5/package.json
@@ -41,7 +41,7 @@
 		"sass": "^1.0.0",
 		"sass-loader": "^13.0.0",
 		"sax": "^1.0.0",
-		"semver": "^7.6.0",
+		"semver": "^7.0.0 || ^6.0.0",
 		"source-map": "^0.7.0",
 		"terser-webpack-plugin": "^5.0.0",
 		"ts-dedent": "^2.0.0",

--- a/packages/webpack5/package.json
+++ b/packages/webpack5/package.json
@@ -41,6 +41,7 @@
 		"sass": "^1.0.0",
 		"sass-loader": "^13.0.0",
 		"sax": "^1.0.0",
+		"semver": "^7.6.0",
 		"source-map": "^0.7.0",
 		"terser-webpack-plugin": "^5.0.0",
 		"ts-dedent": "^2.0.0",

--- a/packages/webpack5/src/configuration/angular.ts
+++ b/packages/webpack5/src/configuration/angular.ts
@@ -3,7 +3,7 @@ import Config from 'webpack-chain';
 import { existsSync } from 'fs';
 
 import { getTypescript, readTsConfig } from '../helpers/typescript';
-import { getDependencyPath } from '../helpers/dependencies';
+import { getDependencyVersion } from '../helpers/dependencies';
 import { getProjectTSConfigPath } from '../helpers/project';
 import { env as _env, IWebpackEnv } from '../index';
 import { warnOnce } from '../helpers/log';
@@ -13,6 +13,7 @@ import {
 	getPlatformName,
 } from '../helpers/platform';
 import base from './base';
+import { satisfies } from 'semver';
 
 // until we can switch to async/await on the webpack config, copy this from '@angular/compiler-cli'
 const GLOBAL_DEFS_FOR_TERSER = {
@@ -187,13 +188,16 @@ export default function (config: Config, env: IWebpackEnv = _env): Config {
 				.loader('angular-hot-loader');
 		});
 
-		const buildAngularMajorVersion = getBuildAngularMajorVersion();
-		if (buildAngularMajorVersion) {
+		const buildAngularVersion = getDependencyVersion(
+			'@angular-devkit/build-angular'
+		);
+
+		if (buildAngularVersion) {
 			const buildAngularOptions: any = {
 				aot: !disableAOT,
 			};
 
-			if (buildAngularMajorVersion < 15) {
+			if (satisfies(buildAngularVersion, '<15.0.0')) {
 				const tsConfig = readTsConfig(tsConfigPath);
 				const { ScriptTarget } = getTypescript();
 				buildAngularOptions.scriptTarget =
@@ -315,27 +319,6 @@ function getAngularWebpackPlugin(): any {
 	return AngularWebpackPlugin;
 }
 
-const MAJOR_VER_RE = /^(\d+)\./;
-function getBuildAngularMajorVersion() {
-	const buildAngularPath = getDependencyPath('@angular-devkit/build-angular');
-	if (!buildAngularPath) {
-		return null;
-	}
-
-	try {
-		const buildAngularVersion =
-			require(`${buildAngularPath}/package.json`).version;
-
-		const [_, majorStr] = buildAngularVersion.match(MAJOR_VER_RE);
-
-		return Number(majorStr);
-	} catch (err) {
-		// ignore
-	}
-
-	return null;
-}
-
 function tryRequireResolve(path: string) {
 	try {
 		return require.resolve(path);
@@ -348,10 +331,10 @@ function getWebpackLoaderPath() {
 	return (
 		tryRequireResolve(
 			'@angular-devkit/build-angular/src/babel/webpack-loader'
-		) ||
+		) ??
 		tryRequireResolve(
 			'@angular-devkit/build-angular/src/tools/babel/webpack-loader'
-		) ||
+		) ??
 		// fallback to angular 16.1+
 		'@angular-devkit/build-angular/src/tools/babel/webpack-loader'
 	);

--- a/packages/webpack5/src/configuration/angular.ts
+++ b/packages/webpack5/src/configuration/angular.ts
@@ -1,5 +1,6 @@
 import { extname, resolve } from 'path';
 import Config from 'webpack-chain';
+import { satisfies } from 'semver';
 import { existsSync } from 'fs';
 
 import { getTypescript, readTsConfig } from '../helpers/typescript';
@@ -13,7 +14,6 @@ import {
 	getPlatformName,
 } from '../helpers/platform';
 import base from './base';
-import { satisfies } from 'semver';
 
 // until we can switch to async/await on the webpack config, copy this from '@angular/compiler-cli'
 const GLOBAL_DEFS_FOR_TERSER = {

--- a/packages/webpack5/src/configuration/base.ts
+++ b/packages/webpack5/src/configuration/base.ts
@@ -5,6 +5,7 @@ import {
 	HotModuleReplacementPlugin,
 } from 'webpack';
 import Config from 'webpack-chain';
+import { satisfies } from 'semver';
 import { existsSync } from 'fs';
 
 import ForkTsCheckerWebpackPlugin from 'fork-ts-checker-webpack-plugin';
@@ -12,11 +13,11 @@ import { BundleAnalyzerPlugin } from 'webpack-bundle-analyzer';
 import TerserPlugin from 'terser-webpack-plugin';
 
 import { getProjectFilePath, getProjectTSConfigPath } from '../helpers/project';
+import { getDependencyVersion, hasDependency } from '../helpers/dependencies';
 import { PlatformSuffixPlugin } from '../plugins/PlatformSuffixPlugin';
 import { applyFileReplacements } from '../helpers/fileReplacements';
 import { addCopyRule, applyCopyRules } from '../helpers/copyRules';
 import { WatchStatePlugin } from '../plugins/WatchStatePlugin';
-import { hasDependency } from '../helpers/dependencies';
 import { applyDotEnvPlugin } from '../helpers/dotEnv';
 import { env as _env, IWebpackEnv } from '../index';
 import { getValue } from '../helpers/config';
@@ -86,7 +87,7 @@ export default function (config: Config, env: IWebpackEnv = _env): Config {
 		const sourceMapAbsolutePath = getProjectFilePath(
 			`./${
 				env.buildPath ?? 'platforms'
-			}/${platform}-sourceMaps/[file].map[query]`,
+			}/${platform}-sourceMaps/[file].map[query]`
 		);
 		const sourceMapRelativePath = relative(outputPath, sourceMapAbsolutePath);
 		config.output.sourceMapFilename(sourceMapRelativePath);
@@ -272,7 +273,7 @@ export default function (config: Config, env: IWebpackEnv = _env): Config {
 	const configFile = tsConfigPath
 		? {
 				configFile: tsConfigPath,
-			}
+		  }
 		: undefined;
 
 	// set up ts support
@@ -451,7 +452,7 @@ export default function (config: Config, env: IWebpackEnv = _env): Config {
 			 * +-----------------------------------------------------------------------------------------+
 			 */
 			/System.import\(\) is deprecated/,
-		]),
+		])
 	);
 
 	// todo: refine defaults
@@ -516,7 +517,11 @@ export default function (config: Config, env: IWebpackEnv = _env): Config {
 
 function shouldIncludeInspectorModules(): boolean {
 	const platform = getPlatformName();
-	// todo: check if core modules are external
-	// todo: check if we are testing
-	return platform === 'ios' || platform === 'android';
+	const coreVersion = getDependencyVersion('@nativescript/core');
+
+	if (coreVersion && satisfies(coreVersion, '>=8.7.0')) {
+		return platform === 'ios' || platform === 'android';
+	}
+
+	return platform === 'ios';
 }

--- a/packages/webpack5/src/helpers/dependencies.ts
+++ b/packages/webpack5/src/helpers/dependencies.ts
@@ -47,3 +47,23 @@ export function getDependencyPath(dependencyName: string): string | null {
 		return null;
 	}
 }
+
+/**
+ * Utility to get the version of a dependency.
+ *
+ * @param dependencyName
+ * @returns string | null - version of the dependency or null if not found
+ */
+export function getDependencyVersion(dependencyName: string): string | null {
+	const dependencyPath = getDependencyPath(dependencyName);
+	if (!dependencyPath) {
+		return null;
+	}
+
+	try {
+		return require(`${dependencyPath}/package.json`).version;
+	} catch (err) {
+		// ignore
+	}
+	return null;
+}

